lib/block-supports/layout.php

```diff
@@ -89,8 +93,18 @@ function gutenberg_get_layout_style( $selector, $layout, $has_block_gap_support
 				$gap_value = isset( $gap_value['top'] ) ? $gap_value['top'] : null;
 			}
 			if ( $gap_value && ! $should_skip_gap_serialization ) {
-				$style .= "$selector > * { margin-block-start: 0; margin-block-end: 0; }";
-				$style .= "$selector > * + * { margin-block-start: $gap_value; margin-block-end: 0; }";
+				$store->add_rule( "$selector > *" )->set_declarations(
+					array(
+						'margin-block-start' => '0',
```

Comment:
Just to confirm: will we need [this core patch](https://github.com/WordPress/wordpress-develop/pull/2928) to land first in order for `gap`, `margin-block-start` et. al. to pass through `safecss_filter_attr()` unmolestered?

Reply:
We shouldn't need the core patch to be merged, because #40875 included those properties in the `safe_style_css` filter in: https://github.com/WordPress/gutenberg/blob/dcddd9148bb6769afac4bd341979ab924184f098/lib/compat/wordpress-6.1/blocks.php#L16

We'll need it to land before the 6.1 Beta, though 🙂